.github/workflows/release.yml

```diff
@@ -0,0 +1,167 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - 'main'
+      - 'master'
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+env:
+  ITCH_USERNAME: ADDON_NAME_ITCH_USER_NAME
+  ITCH_GAME_ID: ADDON_NAME_ITCH_GAME_ID
+  ASSET_LIB_USERNAME: ADDON_NAME_ASSET_LIB_USER_NAME
+  ASSET_LIB_PASSSWORD: ${{ secrets.ASSET_STORE_PASSWORD }}
+  ASSET_LIB_ASSET_ID: ADDON_NAME_ASSET_LIB_ASSET_ID
+
+jobs:
+  prerelease:
+      if: "github.event.head_commit.message != 'Release preparation: storing Version and Changelog'"
+      name: Preparing release
+      runs-on: ubuntu-latest
+      steps:
+          - uses: actions/checkout@v3
+            with:
+              fetch-depth: 0
+              token: ${{ secrets.PAT }}
+
+          - name: Calculate version
+            id: calculate_version
+            uses: mathieudutour/github-tag-action@v6.0
+            with:
+              tag_prefix: ""
+              github_token: ${{ secrets.PAT }}
+              dry_run: true
+              default_bump: false
+              custom_release_rules: |
+                  fix:patch:Bug Fixes,
+                  hotfix:patch:Bug Fixes,
+                  feature:minor:Features,
+                  change:minor:Changes,
+                  breaking:major:Breaking Changes,
+                  major:major:Major Changes,
+
+          - name: "Write changelog"
+            run: |
+              CHANGELOG=""
+              if [ -e addons/ADDON_NAME/CHANGELOG.md ]
+              then
+                  CHANGELOG=$(cat addons/ADDON_NAME/CHANGELOG.md)
+              fi
+              echo -e "${{steps.calculate_version.outputs.changelog}}\n\n${CHANGELOG}" > addons/ADDON_NAME/CHANGELOG.md
+            if: github.ref == 'refs/heads/main' && steps.calculate_version.outputs.new_version != ''
+
+          - name: "Update version in plugin.cfg"
+            run: |
+              sed -i -re 's/version=".+$/version="${{ steps.calculate_version.outputs.new_version }}"/gi' addons/ADDON_NAME/plugin.cfg
+            if: steps.calculate_version.outputs.new_version != ''
+
+          - name: "Commit"
+            uses: EndBug/add-and-commit@v7.2.1
+            with:
+              message: 'Release preparation: storing Version and Changelog'
+              push: true
+            if: steps.calculate_version.outputs.new_version != ''
+
+  release:
+      # Create release only after the version was bumped and release notes added:
+    if: "github.event.head_commit.message == 'Release preparation: storing Version and Changelog'"
+    runs-on: ubuntu-latest
+
+    steps:
+
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Is `GITHUB_TOKEN` the same as `PAT` or a different one?

Reply:
No, but similar. Added a section to clarify the use pf the `PAT`

Reply:
Removed usage of PAT and updated docs